compiler.rst

```diff
@@ -369,25 +378,34 @@ bytecode step of the compiler.  Several pieces of code throughout Python depend
 on having correct information about what bytecode exists.
 
 First, you must choose a name and a unique identifier number.  The official
-list of bytecode can be found in :file:`Include/opcode.h`.  If the opcode is to
+list of bytecode can be found in :file:`Lib/opcode.py`.  If the opcode is to
 take an argument, it must be given a unique number greater than that assigned to
-``HAVE_ARGUMENT`` (as found in :file:`Include/opcode.h`).
+``HAVE_ARGUMENT`` (as found in :file:`Lib/opcode.py`).
 
-Once the name/number pair has been chosen and entered in Include/opcode.h,
-you must also enter it into :file:`Lib/opcode.py` and
-:file:`Doc/library/dis.rst`.
+Once the name/number pair has been chosen and entered in :file:`Lib/opcode.py`,
+you must also enter it into :file:`Doc/library/dis.rst`, and regenerate
+:file:`Include/opcode.h` and :file:`Python/opcode_targets.h` by running
+``make regen-opcode regen-opcode-targets``.
 
 With a new bytecode you must also change what is called the magic number for
-.pyc files.  The variable ``MAGIC`` in :file:`Python/import.c` contains the
-number.
-Changing this number will lead to all .pyc files with the old ``MAGIC``
-to be recompiled by the interpreter on import.
+.pyc files.  The variable ``MAGIC_NUMBER`` in
+:file:`Lib/importlib/_bootstrap_external.py` contains the number.
+Changing this number will lead to all .pyc files with the old ``MAGIC_NUMBER``
+to be recompiled by the interpreter on import.  Whenever ``MAGIC_NUMBER`` is
+changed, the ranges in the ``magic_values`` array in :file:`PC/launcher.c`
+must also be updated.  Changes to :file:`Lib/importlib/_bootstrap_external.py`
+will take effect only after running ``make regen-importlib``.
 
 Finally, you need to introduce the use of the new bytecode.  Altering
 :file:`Python/compile.c` and :file:`Python/ceval.c` will be the primary places
-to change. But you will also need to change the 'compiler' package.
-The key files to do that are :file:`Lib/compiler/pyassem.py` and
-:file:`Lib/compiler/pycodegen.py`.
+to change. You must to add the case for a new opcode into the 'switch'
```

Comment:
Extra word "to": "You must add the case for a new opcode ...".

Reply:
Thank you! Fixed.